pkgs/development/compilers/llvm/3.7/libc++/default.nix

```diff
@@ -14,7 +14,10 @@ stdenv.mkDerivation rec {
     cmakeFlagsArray=($cmakeFlagsArray -DLIBCXX_CXX_ABI_INCLUDE_PATHS="$NIX_BUILD_TOP/libcxxabi-${version}.src/include")
   '';
 
-  patches = [ ./darwin.patch ];
+  patches = [
+    ./darwin.patch
+    ./r242056.patch
```

Comment:
This is already fixed in 3.8. Fetchpatch didn't work for me, it complained about some filename. Also I had to manually edit paths in the original diff to change them from ```libcxx/trunk/...``` to ```a/...``` and ```b/...``` for compatibility with ```patch -p1```.